catroid/src/test/java/org/catrobat/catroid/test/content/actions/SetThreadColorActionTest.kt

```diff
@@ -0,0 +1,128 @@
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class SetThreadColorActionTest {
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        initializeStaticSingletonMethods()
+        sprite = Sprite("testSprite")
+        createTestableSetThreadColorAction(
+            sprite,
+            SequenceAction(), Formula(COLOR_TEST_VALUE_BLACK)
+        )?.act(1.0f)
+    }
+
+    @Test
+    fun testNormalBehaviorColorChange() {
+        assertEquals(Color.BLACK, sprite?.embroideryThreadColor)
+        createTestableSetThreadColorAction(
+            sprite, SequenceAction(),
+            Formula(COLOR_TEST_VALUE_WHITE)
+        )?.act(1.0f)
+        assertEquals(Color.WHITE, sprite?.embroideryThreadColor)
+        createTestableSetThreadColorAction(
+            sprite, SequenceAction(),
+            Formula(COLOR_TEST_VALUE_BLUE)
+        )?.act(1.0f)
+        assertEquals(Color.BLUE, sprite?.embroideryThreadColor)
+    }
+
+    @Test(expected = NullPointerException::class)
+    fun testNullSprite() {
+        val action = sprite?.actionFactory?.createSetThreadColorAction(
+            null,
+            SequenceAction(), Formula(COLOR_TEST_VALUE_BLACK)
+        )
+        action?.act(1.0f)
+    }
+
+    @Test
+    fun testNullFormula() {
+        createTestableSetThreadColorAction(sprite, SequenceAction(), null)?.act(1.0f)
+        assertEquals(COLOR_INITIALIZED_VALUE, sprite?.embroideryThreadColor)
+    }
+
+    @Test(expected = StringIndexOutOfBoundsException::class)
+    fun testTooShortFormula() {
+        createTestableSetThreadColorAction(
+            sprite,
+            SequenceAction(), Formula("#000")
+        )?.act(1.0f)
+    }
+
+    @Test(expected = NumberFormatException::class)
+    fun testInvalidColorCode() {
+        createTestableSetThreadColorAction(
+            sprite,
+            SequenceAction(), Formula(TEST_INVALID_VALUE)
+        )?.act(1.0f)
+        assertEquals(COLOR_INITIALIZED_VALUE, sprite?.embroideryThreadColor)
+    }
+
+    private fun createTestableSetThreadColorAction(sprite: Sprite?, sequence: SequenceAction?, color: Formula?): Action? {
+        val action = Actions.action(
+            TestableSetThreadColorAction::class.java
+        )
+        val scope = sprite?.let {
+            Scope(ProjectManager.getInstance().currentProject, it, sequence)
+        }
+        action.setSprite(sprite)
+        action.setScope(scope)
+        action.setColor(color)
+        return action
+    }
+
+    class TestableSetThreadColorAction : SetThreadColorAction() {
+        override fun argbToInt(redInt: Int, greenInt: Int, blueInt: Int) =
+            COLOR_ALPHA and 0xff shl 24 or (redInt and 0xff shl 16) or (greenInt and 0xff shl 8) or (blueInt and 0xff)
+    }
+
+    companion object {
+        private var sprite: Sprite? = null
```

Comment:
Please move this out of the companion object. It should be a simple class field. Otherwise all instances of this test class will share the same sprite and might influence each other -> flaky tests.